flink-core/src/main/java/org/apache/flink/configuration/StateBackendOptions.java

```diff
@@ -88,4 +88,35 @@
                     .defaultValue(128)
                     .withDescription(
                             "Defines the number of measured latencies to maintain at each state access operation.");
+
+    @Documentation.Section(Documentation.Sections.STATE_BACKEND_MATERIALIZATION)
+    public static final ConfigOption<Long> PERIODIC_MATERIALIZATION_INTERVAL =
+            ConfigOptions.key("state.backend.periodic-materialize.interval")
+                    .longType()
+                    .defaultValue(10000L)
+                    .withDescription(
+                            "Defines the interval in milliseconds to perform periodic materialization for state backend.");
+
+    @Documentation.Section(Documentation.Sections.STATE_BACKEND_MATERIALIZATION)
+    public static final ConfigOption<Long> PERIODIC_MATERIALIZATION_INIT_DELAY =
+            ConfigOptions.key("state.backend.periodic-materialize.initial-delay")
+                    .longType()
+                    .defaultValue(10000L)
```

Comment:
I think this default is too small if we are targeting 500-1000ms checkpoint duration.
"Continuous materialization" is also an option but maybe we should start with the default of `execution.checkpointing.timeout` - 10min?